contracts/XC20Safe.sol

```diff
@@ -0,0 +1,37 @@
+// The Licensed Work is (c) 2022 Sygma
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.11;
+
+import "./ERC20Safe.sol";
+import "./interfaces/IERC20Plus.sol";
+
+/**
+    @title Manages deposited XC20s.
+    @author ChainSafe Systems.
+    @notice This contract is intended to be used with XC20Handler contract.
+ */
+contract XC20Safe is ERC20Safe {
+
+    /**
+        @notice Used to create new XC20s.
+        @param tokenAddress Address of XC20 to transfer.
+        @param recipient Address to mint token to.
+        @param amount Amount of token to mint.
+     */
+    function mintERC20(address tokenAddress, address recipient, uint256 amount) internal override {
```

Comment:
should the names be mintXC20 and burnXC20? but since you inheriting ERC20Safe and try to override, I guess they have to be mintERC20 and burnERC20 in this case, but IMO these names might cause confusion because you also call them from xc20handler

Reply:
I think this should stay as is to comply with inheritance, also methods are internal so there shouldn't be user confusion